test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,191 @@
+		framework.ExpectEqual(event.Note, eventPatchNote, "test event note does not match patch note")
+
+		ginkgo.By("deleting the test event")
+		// delete original event
+		err = f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).Delete(context.TODO(), eventTestName, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "failed to delete the test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespace
+		eventsList, err = f.ClientSet.EventsV1beta1().Events("").List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "testevent-constant=true",
+		})
+		framework.ExpectNoError(err, "fail to list all events")
+		foundCreatedEvent = false
+		for _, val := range eventsList.Items {
+			if val.ObjectMeta.Name == eventTestName && val.ObjectMeta.Namespace == f.Namespace.Name {
+				foundCreatedEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+	})
+
+	ginkgo.It("should delete a collection of events", func() {
+		eventTestNames := []string{"test-event-1", "test-event-2", "test-event-3"}
+		someTime := metav1.MicroTime{Time: time.Unix(1505828956, 0)}
+
+		ginkgo.By("Create set of events")
+		// create a test event in test namespace
+		for _, eventTestName := range eventTestNames {
+			eventNote := "This is " + eventTestName
+			_, err := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).Create(context.TODO(), &v1beta1.Event{
+
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   eventTestName,
+					Labels: map[string]string{"testevent-set": "true"},
+				},
+				EventTime:           someTime,
+				Note:                eventNote,
+				Reason:              "Test",
+				Type:                "Normal",
+				ReportingController: "test-controller",
+				ReportingInstance:   "test-node",
+			}, metav1.CreateOptions{})
+			framework.ExpectNoError(err, "failed to create event")
+			framework.Logf("created %v", eventTestName)
+		}
+
+		ginkgo.By("get a list of Events with a label in the current namespace")
+		// get a list of events
+		eventList, err := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "testevent-set=true",
+		})
+		framework.ExpectNoError(err, "failed to get a list of events")
+		framework.ExpectEqual(len(eventList.Items), len(eventTestNames), "event list should have 3 items")
+
+		ginkgo.By("delete a list of events")
+		// delete a list of events
+		framework.Logf("requesting DeleteCollection of events")
+		err = f.ClientSet.EventsV1beta1().Events(f.Namespace.Name).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: "testevent-set=true"})
+		framework.ExpectNoError(err, "failed to delete the test event")
+
+		ginkgo.By("check that the list of events matches the requested quantity")
+		err = wait.PollImmediate(eventRetryPeriod, eventRetryTimeout, checkEventListQuantity(f, "testevent-set=true", 0))
```

Comment:
instead of introducing a dedicate function, maybe just:

```
pollFn := func() (bool, error) {
  eventList, err := client.Events(f.Namespace.Name).List(context.TODO(), ...)
  if err != nil {
    return false, nil
  }
  return len(eventList.Items) == quantity, nil
}
```

Reply:
Done.